datamol/mol.py

```diff
@@ -85,6 +85,22 @@ def to_mol(
     return _mol
 
 
+def same_mol(mol1: Optional[Chem.rdchem.Mol], mol2: Optional[Chem.rdchem.Mol]):
+    """Check two molecules are the same by comparing their RDKit canonical SMILES.
+
+    Invalid molecules (None) are always considered as not the same.
+
+    Args:
+        mol1: A molecule.
+        mol2: A molecule.
+    """
+
+    if mol1 is None or mol2 is None:
+        return False
+
+    return dm.to_smiles(mol1, canonical=True) == dm.to_smiles(mol2, canonical=True)
```

Comment:
with the new rdkit kekulization algo, this would fail when mol are not sanitized first.

```
mol1 = dm.to_mol("C1=CC=CC=C1", sanitize=False)
Chem.Kekulize(mol1)
dm.to_smiles(mol1, canonical=True) # ===> C1=CC=CC=C1 not the regular benzene c1ccccc1
```

dm.inchi_key is better imo

Reply:
Even without this kekulization issue, InChiKey is actually better and more appropriate for the job.